### PR TITLE
Uuid support in mock cluster

### PR DESCRIPTION
**Pull request by emasab**

### CODE REVIEW

File: src/rdkafka.c [hunk 1]
```diff
@@ -5077,6 +5077,38 @@ rd_kafka_Uuid_t *rd_kafka_Uuid_copy(const rd_kafka_Uuid_t *uuid) {
...
+                 * a random value. */
+                rand_values_app = (unsigned char *)&rand_uint16;
+                rand_values_bytes[i] |= rand_values_app[0];
+                rand_values_bytes[i + 1] |= rand_values_app[1];
+        }
+
+        rand_values_bytes[6] &= 0x0f; /* clear version */
+        rand_values_bytes[6] |= 0x40; /* version 4 */
+        rand_values_bytes[8] &= 0x3f; /* clear variant */
+        rand_values_bytes[8] |= 0x80; /* IETF variant */
```
Comment (pranavrth) [inline src/rdkafka.c:5105]: I think we should move this to rdrand.h

File: src/rdkafka.c [hunk 2]
```diff
@@ -5077,6 +5077,38 @@ rd_kafka_Uuid_t *rd_kafka_Uuid_copy(const rd_kafka_Uuid_t *uuid) {
...
 }
 
+/**
+ * Returns a new non cryptographically secure UUIDv4 (random).
+ *
+ * @return A UUIDv4.
+ *
+ * @remark Must be freed after use using rd_kafka_Uuid_destroy().
+ */
+rd_kafka_Uuid_t *rd_kafka_Uuid_random() {
```
Comment (pranavrth) [inline src/rdkafka.c:5087]: Why are we returning a pointer instead of a value here?

File: src/rdkafka.c (hunk 1 above)
Comment (emasab) [reply src/rdkafka.c:5105]: It generates a Uuid, all the code for a Uuid is here

File: src/rdkafka.c [hunk 3]
```diff
@@ -5086,6 +5118,40 @@ void rd_kafka_Uuid_destroy(rd_kafka_Uuid_t *uuid) {
...
+ * @brief Computes canonical encoding for the given uuid string.
+ *        Mainly useful for testing.
+ *
+ * @param uuid UUID for which canonical encoding is required.
+ *
+ * @return canonical encoded string for the given UUID.
+ *
+ * @remark  Must be freed after use.
+ */
+const char *rd_kafka_Uuid_str(const rd_kafka_Uuid_t *uuid) {
```
[the remark this answers is not in the document]
Comment (emasab) [reply src/rdkafka.c:5131]: It prints the UUID in its standard format, it's useful to check if the UUID respects the standard v4 (the 4 always in the same position and the [89ab] in next sequence.

File: src/rdkafka.c (hunk 1 above)
Comment (emasab) [reply src/rdkafka.c:5105]: We should refactor it to a `rduuid.h` + `rduuid.c` later

Comment (emasab) [reply src/rdkafka.c:5105]: Maybe I can move it to a `rduuid.h` and `rduuid.c`

Comment (pranavrth) [reply src/rdkafka.c:5105]: I was talking about the generic part related to random bytes generation here.

File: src/rdkafka.c [hunk 4]
```diff
@@ -5077,6 +5077,38 @@ rd_kafka_Uuid_t *rd_kafka_Uuid_copy(const rd_kafka_Uuid_t *uuid) {
...
 }
 
+/**
+ * Returns a new non cryptographically secure UUIDv4 (random).
+ *
+ * @return A UUIDv4.
+ *
+ * @remark Must be freed after use using rd_kafka_Uuid_destroy().
+ */
+rd_kafka_Uuid_t rd_kafka_Uuid_random() {
```
Comment (pranavrth) [inline src/rdkafka.c:5087]: Move this to rdkafka_proto.h where all the other internal functions related to Uuid are present

File: src/rdkafka.c (hunk 3 above)
Comment (pranavrth) [inline src/rdkafka.c:5131]: Move this to rdkafka_proto.h where all the other internal functions related to Uuid are present

File: src/rdkafka.c (hunk 1 above)
Comment (pranavrth) [reply src/rdkafka.c:5105]: Regarding rdkafka_uuid.h and rdkafka_uuid.c, we can have them but don't do it for now. Let's do it later when we we won't have much conflicts with other ongoing dev work.

**Review by pranavrth — CHANGES_REQUESTED** [review]

File: src/rdkafka.c (hunk 4 above)
Comment (emasab) [reply src/rdkafka.c:5087]: that is just a header we cannot move all the implementation there, we can refactor to its own file

File: src/rdkafka.c (hunk 3 above)
Comment (emasab) [reply src/rdkafka.c:5131]: same

File: src/rdkafka.c (hunk 1 above)
Comment (emasab) [reply src/rdkafka.c:5105]: > I think we should move this to rdrand.h
here the random value is generated in `rd_jitter` the rest is code related to Uuid

**Review by pranavrth — APPROVED** [review]
LGTM!

**Pull request merged**